middle_end/flambda2.0/types/template/flambda_type.templ.ml

```diff
@@ -281,7 +281,8 @@ let prove_is_int env t : bool proof =
         if is_bottom env imms then Proved false
         else Unknown
     end
-  | Value (Ok _) -> Invalid
+  | Value (Ok (Boxed_float _ | Boxed_int32 _ | Boxed_int64 _
+      | Boxed_nativeint _ | Closures _ | String _ | Array _)) -> Proved true
```

Comment:
Shouldn't that be `Proved false`, considering that we are checking `is_int` and we seem to have a boxed value ?

Reply:
Whoops, well spotted.